File: api-master-data/src/entities/product-note-assignment.entity.ts
import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  ManyToOne,
  JoinColumn,
  Index,
  Unique,
} from 'typeorm';
import { Product } from './product.entity';
import { ProductNote } from './product-note.entity';

@Entity('product_note_assignments')
@Index(['tenantId', 'productId'])
@Unique(['productId', 'noteId'])
export class ProductNoteAssignment {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @Column({ name: 'tenant_id' })
  tenantId: string;

  @Column({ name: 'product_id' })
  productId: string;

  @ManyToOne(() => Product, { onDelete: 'CASCADE' })
  @JoinColumn({ name: 'product_id' })
  product: Product;

  @Column({ name: 'note_id' })
  noteId: string;

  @ManyToOne(() => ProductNote, { onDelete: 'CASCADE' })
  @JoinColumn({ name: 'note_id' })
  note: ProductNote;

  @Column({ name: 'sort_order', default: 0 })
  sortOrder: number;
}

File: ccb-android/app/src/main/java/com/techres/ccb/data/local/entity/ProductNoteAssignmentEntity.kt
package com.techres.ccb.data.local.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index

@Entity(
    tableName = "product_note_assignments",
    primaryKeys = ["product_id", "note_id"],
    indices = [
        Index(value = ["note_id"]),
        Index(value = ["product_id"])
    ],
    foreignKeys = [
        ForeignKey(
            entity = ProductNoteEntity::class,
            parentColumns = ["id"],
            childColumns = ["note_id"],
            onDelete = ForeignKey.CASCADE
        )
    ]
)
data class ProductNoteAssignmentEntity(
    @ColumnInfo(name = "product_id")
    val productId: String,

    @ColumnInfo(name = "note_id")
    val noteId: String,

    @ColumnInfo(name = "branch_id")
    val branchId: String,

    @ColumnInfo(name = "sort_order")
    val sortOrder: Int = 0
)
